lib/mongo/encoder.ex

```diff
@@ -0,0 +1,10 @@
+defprotocol Mongo.Encoder do
+  @fallback_to_any false
+
+  @spec encode(t) :: map()
+  def encode(value)
+end
+
+defimpl Mongo.Encoder, for: Map do
+  def encode(v), do: v
+end
```

Comment:
Is this necessary? The encoder code checks the existing of an implementation: `if Mongo.Encoder.impl_for(value)`, so I guess, we need not an implementation for a map?